### PR TITLE
Docs: Migrate sharing_declarations.rst to pure python mode

### PR DESCRIPTION
**Pull request by matusvalo**
Part of #4187.

### CODE REVIEW

File: docs/src/userguide/sharing_declarations.rst [hunk 1]
```diff
@@ -193,32 +240,45 @@ Python methods.
...
-.. literalinclude:: ../../examples/userguide/sharing_declarations/setup.py
+        .. literalinclude:: ../../examples/userguide/sharing_declarations/setup_pyx.py
+            :caption: setup.py
 
 Some things to note about this example:
 
-* There is a :keyword:`cdef` class Shrubbery declaration in both
-  :file:`Shrubbing.pxd` and :file:`Shrubbing.pyx`. When the Shrubbing module
+* There is a :keyword:`cdef`/``@cclass`` class Shrubbery declaration in both
+  :file:`Shrubbing.pxd` and :file:`Shrubbing.pyx`. When the shrubbing module
```
Comment (da-woods) [inline docs/src/userguide/sharing_declarations.rst:277]: Capitalization here? (Isn't consistent with below)

File: docs/src/userguide/sharing_declarations.rst [hunk 2]
```diff
@@ -193,32 +240,45 @@ Python methods.
...
   is compiled, these two declarations are combined into one.
-* In Landscaping.pyx, the :keyword:`cimport` Shrubbing declaration allows us
-  to refer to the Shrubbery type as :class:`Shrubbing.Shrubbery`. But it
-  doesn't bind the name Shrubbing in Landscaping's module namespace at run
+* In :file:`landscaping.pyx`/:file:`landscaping.py`, the :keyword:`cimport` shrubbing
+  declaration allows us to refer to the Shrubbery type as :class:`Shrubbing.Shrubbery`.
```
Comment (da-woods) [inline docs/src/userguide/sharing_declarations.rst:280]: More capitalization

File: docs/src/userguide/sharing_declarations.rst [hunk 3]
```diff
@@ -193,32 +240,45 @@ Python methods.
...
+  But it doesn't bind the name Shrubbing in Landscaping's module namespace at run
   time, so to access :func:`Shrubbing.standard_shrubbery` we also need to
   ``import Shrubbing``.
```
Comment (da-woods) [inline docs/src/userguide/sharing_declarations.rst:283]: And here...

File: docs/src/userguide/sharing_declarations.rst [hunk 4]
```diff
@@ -235,7 +295,7 @@ Versioning
 ``.pxd`` files can be labelled with a minimum Cython version as part of
 their file name, similar to the version tagging of ``.so`` files in PEP 3149.
 For example a file called :file:`Shrubbing.cython-30.pxd` will only be
```
Comment (da-woods) [inline docs/src/userguide/sharing_declarations.rst:297]: And here

**Comment by matusvalo** [comment]
Fixed. All modules were lowercased to match the code examples.

File: docs/src/userguide/sharing_declarations.rst [hunk 5]
```diff
@@ -193,34 +240,47 @@ Python methods.
...
-.. literalinclude:: ../../examples/userguide/sharing_declarations/shrubbing.pyx
+        .. literalinclude:: ../../examples/userguide/sharing_declarations/setup_py.py
+            :caption: setup.py
 
-:file:`landscaping.pyx`:
+    .. group-tab:: Cython
 
-.. literalinclude:: ../../examples/userguide/sharing_declarations/landscaping.pyx
+        .. literalinclude:: ../../examples/userguide/sharing_declarations/shrubbing.py
+            :caption: shrubbing.py
```
Comment (da-woods) [inline docs/src/userguide/sharing_declarations.rst:264]:
```suggestion
        .. literalinclude:: ../../examples/userguide/sharing_declarations/shrubbing.pyx
            :caption: shrubbing.pyx
```

You have the pure Python and the Cython files the wrong way around here

File: docs/src/userguide/sharing_declarations.rst [hunk 6]
```diff
@@ -193,34 +240,47 @@ Python methods.
...
+
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/userguide/sharing_declarations/shrubbing.pyx
+            :caption: shrubbing.pyx
+
+        .. literalinclude:: ../../examples/userguide/sharing_declarations/landscaping.pyx
+            :caption: landscaping.pyx
```
Comment (da-woods) [inline docs/src/userguide/sharing_declarations.rst:254]:
```suggestion
        .. literalinclude:: ../../examples/userguide/sharing_declarations/shrubbing.py
            :caption: shrubbing.py

        .. literalinclude:: ../../examples/userguide/sharing_declarations/landscaping.py
            :caption: landscaping.py
```

File: docs/src/userguide/sharing_declarations.rst [hunk 7]
```diff
@@ -193,34 +240,47 @@ Python methods.
...
-:file:`landscaping.pyx`:
+    .. group-tab:: Cython
 
-.. literalinclude:: ../../examples/userguide/sharing_declarations/landscaping.pyx
+        .. literalinclude:: ../../examples/userguide/sharing_declarations/shrubbing.py
+            :caption: shrubbing.py
 
-One would then need to compile both of these modules, e.g. using
+        .. literalinclude:: ../../examples/userguide/sharing_declarations/landscaping.py
+            :caption: landscaping.py
```
Comment (da-woods) [inline docs/src/userguide/sharing_declarations.rst:267]:
```suggestion
        .. literalinclude:: ../../examples/userguide/sharing_declarations/landscaping.pyx
            :caption: landscaping.pyx
```

**Comment by matusvalo** [comment]
Done thank you @da-woods !

**Comment by da-woods** [comment]
Thanks

**Pull request merged**